sympy/solvers/tests/test_solveset.py

```diff
@@ -2490,6 +2491,54 @@ def test_issue_10426():
         Intersection(S.Complexes, ImageSet(Lambda(n, -I*(I*(2*n*pi + arg(-exp(-2*I*x))) + 2*im(x))),
         S.Integers)))).dummy_eq(Dummy('x,n'))
 
+def test_issue_18208():
+    # linsolve
+    vars = symbols('x0:16') + symbols('y0:12')
+    x0, x1, x2, x3, x4, x5, x6, x7, x8, x9, x10, x11, x12, x13, x14, x15,\
+    y0, y1, y2, y3, y4, y5, y6, y7, y8, y9, y10, y11 = vars
+
+    eqs = [x0 + x1 + x2 + x3 - 51, x0 + x1 + x4 + x5 - 46, x2 + x3 + x6 + x7 - 39,\
+           x0 + x3 + x4 + x7 - 50, x1 + x2 + x5 + x6 - 35, x4 + x5 + x6 + x7 - 34,\
+           x4 + x5 + x8 + x9 - 46, x10 + x11 + x6 + x7 - 23, x11 + x4 + x7 + x8 - 25,\
+           x10 + x5 + x6 + x9 - 44, x10 + x11 + x8 + x9 - 35, x12 + x13 + x8 + x9 - 35,\
+           x10 + x11 + x14 + x15 - 29, x11 + x12 + x15 + x8 - 35,  x10 + x13 + x14 + x9 - 29,\
+           x12 + x13 + x14 + x15 - 29, y0 + y1 + y2 + y3 - 55, y0 + y1 + y4 + y5 - 53,\
+           y2 + y3 + y6 + y7 - 56, y0 + y3 + y4 + y7 - 57, y1 + y2 + y5 + y6 - 52,\
+           y4 + y5 + y6 + y7 - 54, y4 + y5 + y8 + y9 - 48, y10 + y11 + y6 + y7 - 60,\
+           y11 + y4 + y7 + y8 - 51, y10 + y5 + y6 + y9 - 57, y10 + y11 + y8 + y9 - 54]
+    partsol = [x10 - 2, x11 - 5, x12 - 1, x13 - 6, x14 - 1, x15 - 21, y0 - 12, y1 - 20]
+
+    # solve
+    solve_expected = {x0: 38 - x3, x1: x3 - 10, x2: 23 - x3, x4: 12 - x7, x5: x7 + 6,\
+                x6: 16 - x7, x8: 8, x9: 20, x10: 2, x11: 5, x12: 1, x13: 6, x14: 1,\
+                x15: 21, y0: 12, y1: 20, y2: -y11 + y9 + 2, y3: y11 - y9 + 21,\
+                y4: -y11 - y7 + y9 + 24, y5: y11 + y7 - y9 - 3, y6: 33 - y7, y8: 27 - y9,\
+                y10: 27 - y11}
+
+    assert solve(eqs+partsol, vars) == solve_expected
+
+    # linsolve
+    matrix = linear_eq_to_matrix(eqs+partsol, vars)
+    linsolve_expected = FiniteSet((38 - x3, x3 - 10, 23 - x3, x3, 12 - x7, x7 + 6, 16 - x7, x7,\
+                          8, 20, 2, 5, 1, 6, 1, 21, 12, 20, -y11 + y9 + 2, y11 - y9 + 21,\
+                          -y11 - y7 + y9 + 24, y11 + y7 - y9 - 3, 33 - y7, y7, 27 - y9, y9,\
+                          27 - y11, y11))
+
+    assert linsolve(eqs+partsol, vars) == linsolve_expected
+    assert linsolve(matrix, vars) == linsolve_expected
+
+    # gauss_jordan_solve
+    lhs, rhs = linear_eq_to_matrix(eqs+partsol, vars)
+    gj_solve, new_vars = lhs.gauss_jordan_solve(rhs)
+    gj_solve = [i for i in gj_solve]
+    tau0, tau1, tau2, tau3, tau4 = symbols([str(v) for v in new_vars])
+
+    gj_expected = [38 - tau0, tau0 - 10, 23 - tau0, tau0, 12 - tau1, tau1 + 6, 16 - tau1,\
+                tau1, 8, 20, 2, 5, 1, 6, 1, 21, 12, 20, tau3 - tau4 + 2, -tau3 + tau4 + 21,\
+                -tau2 + tau3 - tau4 + 24, tau2 - tau3 + tau4 - 3, 33 - tau2, tau2,\
+                27 - tau3, tau3, 27 - tau4, tau4]
```

Comment:
Can we make this expected solution from the other one using subs to replace the symbols?

The idea is to be clear that it is the same solution but just with different symbol names.

Reply:
Like `gj_expected = linsolve_expected.subs(zip([x3, ...], vars))`.

Reply:
Sure. Is there a cleaner way than this?

```py
gj_expected = list(list(linsolve_expected.subs(zip([x3, x7, y7, y9, y11], [tau0, tau1, tau2, tau3, tau4])))[0])
```

Reply:
Does the code I showed not work?

Reply:
Oh sorry, for some reason I didn't see that when I replied.

No, it doesn't seem to work:

```pycon
        gj_expected = linsolve_expected.subs(zip([x3, x7, y7, y9, y11], new_vars))
>       assert gj_solve == gj_expected
E       assert [38 - tau0, t...tau1 + 6, ...] == FiniteSet((38...- tau4, tau4))
E         Use -v to get the full diff

test_solveset.py:2538: AssertionError
```

`linsolve_expected` is a tuple in a FiniteSet, whereas `gj_solve` is just a list.

Reply:
Oh, sorry I see what you mean. Yes, I guess what you have is fine. You can also convert the other way like
```
assert FiniteSet(Tuple(gj_solve)) == gj_expected
```

Reply:
I would break up the long line though. Perhaps do the subs in one line and the other parts in another.

Reply:
Actually converting `gj_solve` looks quite nice. I'll do that.